### ISSUE TITLE
CSharpErrorLog - Error List Build Diagnostics failing due to file write lock

### ISSUE THREAD

**Issue by tannergooding** [dotnet/roslyn#12352]
The test is consistently failing during the second Build/View.ErrorList invocation:

```
17:17:47  Scenario: Error List Build Diagnostics
17:17:47 	Executing action [7/6/2016 12:02:37 AM] Build
17:17:47 	Executing action [7/6/2016 12:02:51 AM] ExecuteCommand
17:17:47 	Executing action [7/6/2016 12:02:51 AM] VerifyErrorLog
17:17:47 	Executing action [7/6/2016 12:03:54 AM] CleanSolution
17:17:47 	Executing action [7/6/2016 12:03:55 AM] Build
17:17:47 	Executing action [7/6/2016 12:04:08 AM] ExecuteCommand
17:17:47 	Executing action [7/6/2016 12:04:08 AM] VerifyErrorLog
17:17:47 --------------------------------
17:17:47 System.Exception: Verify Error log failed:
17:17:47 
17:17:47 Expected Count: '9799'
17:17:47 Actual Count: '9800'
17:17:47 
17:17:47 Details:
17:17:47 
17:17:47 	Unexpected diagnostics: 
17:17:47 		"ruleId": "CS2012",
```

**Comment by tannergooding** [comment]
FYI. @cston

**Comment by jasonmalinowski** [comment]
FYI to @Srivatsn.

**Comment by mavasani** [comment]
@tannergooding Can you point me to Jenkins failures for this test?

**Comment by mavasani** [comment]
Okay so we have 2 error log tests in Microupdate branch, one for v1 diagnostic engine and another one for v2 engine. v1 engine test `CSharpErrorLog_v1_Engine.xml` needs to be just deleted from the Microupdate branch, it has already been deleted in master. v2 engine test `CSharpErrorLog.xml` is currently blocked by https://github.com/dotnet/roslyn/issues/12289.

I'll track this issue to remove the v1 engine test in Microupdate branch.

**Comment by cston** [comment]
@mavasani #12289 should be fixed now.

**Comment by mavasani** [comment]
Okay so this test passes locally for me, I tried executing it multiple times (I synced to the tip of master for both roslyn-internal and Open). I have Dev14 Update3 on my box.

I attempt to enable the test with https://github.com/dotnet/roslyn-internal/pull/1154/files, but it fails on the Jenkins machine due to the above issue.
The failure above seems specific to something locking a file during the first build, causing an extra CS2012 diagnostic to be emitted when a clean/build is done again. The test correctly flags the extra diagnostic as unexpected.

Few questions for @dotnet/roslyn-infrastructure: 

1. When did we start seeing this failure? Do we know what change/date this started failing consistently due to this issue?
2. Does the Jenkins machine where we run VSI tests have Dev14 Update3 build or Dev15 build or both SxS? 
3. What compiler toolset is being used on the Jenkins machine? I see that https://github.com/dotnet/roslyn/issues/10414 has been pushed out, so is it likely we are still using very old compiler toolset for Jenkins runs?

Regardless, this is either an infrastructure issue (Jenkins machine setup) OR a compiler issue (build locking a file, which causes extra CS2012 on next clean/build cycle). I am moving this back to infrastructure team.

**Comment by tannergooding** [comment]
@mavasani, currently VS2013.5 + VS2015.3 SxS... Soon we will have those + VS15.P3 SxS.

We are using v1.3.0 for the bootstrap build and the latest bits for the full build.

**Comment by jasonmalinowski** [comment]
This has still failed multiple times in the last day: http://jdash.azurewebsites.net/builds/test?name=CSharpErrorLog.Error%20List%20Build%20Diagnostics&viewName=dotnet_roslyn&pr=False&limit=10&startDate=2016-07-20

Per @mavasani's analysis, moving over to compiler, since that error is there.

**Comment by natidea** [comment]
@jaredpar Could you get someone to take a look at this?

**Comment by jasonmalinowski** [comment]
@jaredpar @mavasani This is currently the flakiest test we have in our test suite. Can you work together to figure out what the next step is for this?

**Comment by jasonmalinowski** [comment]
Ack, sorry, my bad, this is actually a different bug.

**Comment by mavasani** [comment]
@jasonmalinowski Can you point me to the failure logs? Scenario for the initial issue that Tanner raised has already been [disabled](https://github.com/dotnet/roslyn-internal/blob/master/Closed/Hosting/RoslynTaoActions/IntegrationTests/CSharpErrorLog.xml#L19), so any failures for this test is a different unknown issue and we should track it separately.

**Comment by jasonmalinowski** [comment]
Moving off of infrastructure: let's get the test ported to the new infrastructure and we'll see if it still fails.

**Comment by CyrusNajmabadi** [comment]
Closing out due to ancient test infrastructure.
